### PR TITLE
Remove pow and float_power TestGradient Skips

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76481

### CODE REVIEW

**Comment by drisspg** [comment]
Linked issues: #74279 and #76483

**Comment by drisspg** [comment]
Sorry new to ghstack and accidentally made a new commit instead of ammending this one

**Comment by mruberry** [comment]
Looks like you got an unlucky base with some unrelated test failures, @drisspg -- would you just rebase this please so the tests can run? I've also added the ciflow/all label to run additional jobs typically only run on HUD

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -13045,27 +13045,21 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
-                        DecorateInfo(unittest.skip("Skipped!"), 'TestGradients'),
-                    )),
+                    rhs_make_tensor_kwargs=dict(low=1)),
     BinaryUfuncInfo('float_power',
                     dtypes=all_types_and_complex_and(torch.half, torch.bfloat16, torch.bool),
                     promotes_int_to_float=True,
                     supports_forward_ad=True,
                     supports_fwgrad_bwgrad=True,
                     supports_one_python_scalar=True,
+                    lhs_make_tensor_kwargs=dict(low=0),
```
[the remark this answers is not in the document]
Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py:13058]: @mruberry thoughts?

[the remark this answers is not in the document]
Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py]: Updated with sampling function

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1470,6 +1470,22 @@ def sample_inputs_linalg_matrix_power(op_info, device, dtype, requires_grad, **k
         for n in [-4, -2, -1]:
             yield SampleInput(make_arg_fullrank(*size), args=(n,))
 
+def sample_inputs_pow(op, device, dtype, requires_grad, **kwargs):
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:1473]: Since we're adding a custom sample input function, maybe we want to add a test case that checks the complex + whatever -> complex type promotion case?

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -13144,27 +13160,18 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
...
                     supports_two_python_scalars=True,
-                    # TODO: FIXME: pow needs a way of specifying that for integer
-                    #   types only it does not support negative exponentes
-                    rhs_make_tensor_kwargs=dict(low=1),
-                    skips=(
-                        # nan vs nan comparisons
-                        # https://github.com/pytorch/pytorch/issues/74279
-                        DecorateInfo(unittest.skip("Skipped!"), 'TestGradients'),
-                    )),
+                    sample_inputs_func=sample_inputs_pow),
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:13163]: It would be nice to have the `.ref` field added in this PR. Maybe we could add skips specifically for the cases that are failing?

**Review by soulitzer — COMMENTED** [review]
Looks good so far. I commented about some things we want probably want to add though while we are here.

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py:1473]: By adding the np.power ref I see that I have to skip a test_type_promotion skip. Does that overlay/cover the need for this?

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:1473]: Why does the adding the ref require skipping test_type_promotion?

Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py:1455]: Sorry I worded that wrong, once I added the ref I found that I need to add a skip for test_type_promotion. 

I did some further digging and here: https://github.com/pytorch/pytorch/blob/f6bbecf8b503eb6562d52b66124a49b45a57f14a/test/test_binary_ufuncs.py#L192
Even though a sampling function is passed into the op the default generator is used. So I was able to repass in the rhs kwarg.  However for the other - small, large, extremal these are hard coded special values. I could not figure out a way of specifying to not raise the ints to negative exponents leading to the the skips.  

The complex skips seem to be comparison of nans again. I am not entirely sure if this is caused by overflow/ underflow but they seem to be consistent with numpy.

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -13655,29 +13949,60 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                     supports_fwgrad_bwgrad=True,
                     assert_autodiffed=True,
                     supports_one_python_scalar=True,
-                    # TODO: FIXME: pow needs a way of specifying that for integer
-                    #   types only it does not support negative exponentes
-                    rhs_make_tensor_kwargs=dict(low=1),
+                    # Raising integer values to negative values is not causes an error
```
Comment (drisspg) [inline torch/testing/_internal/common_methods_invocations.py:13952]: work is here
